### PR TITLE
Updates to Caribou, Timepix3 and AIDA TLU Producers

### PR DESCRIPTION
**Pull request by simonspa**
## Caribou

* Improve detection of 2nd T0 signal when decoding data

## Timepix3

* Detect multiple T0s
* Allow configuration of the decoder

## AIDA TLU

* Read and interpret `fineTS` flags (fine timestamps from AIDA TLU)
* Allow calibration of time signal using trigger input coincidence and fine timestamps
* Allow setting for the trigger mask to select which trigger inputs to be used for the timestamp calculation

### CODE REVIEW

File: user/tlu/misc/aida_tlu/aida_tlu_connection.xml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <connections>
     <connection id="aida_tlu.controlhub" uri="chtcp-2.0://localhost:10203?target=192.168.200.30:50001"
-        address_table="file://./aida_tlu_address-fw_version_14.xml" />
+        address_table="file://./aida_tlu_address-fw_version_25.xml" />
```
Comment (lhuth) [inline user/tlu/misc/aida_tlu/aida_tlu_connection.xml:4]: This is not the default firmware yet (mainly a ping to @herkerta ). Also an important message in the upcoming tag

File: user/tlu/module/src/TluRawEvent2StdEventConverter.cc [hunk 2]
```diff
@@ -29,41 +29,142 @@ bool TluRawEvent2StdEventConverter::Converting(eudaq::EventSPC d1, eudaq::Standa
...
-  auto finets = ((finets0 * ((triggersFired >> 0) & 0x1))
-               + (finets1 * ((triggersFired >> 1) & 0x1))
-               + (finets2 * ((triggersFired >> 2) & 0x1))
-               + (finets3 * ((triggersFired >> 3) & 0x1))
-               + (finets4 * ((triggersFired >> 4) & 0x1))
-               + (finets5 * ((triggersFired >> 5) & 0x1)))
-               / __builtin_popcount(triggersFired & 0x3F); // count "ones" in binary
-
-  auto ts = (d1->GetTimestampBegin() * 1000) & 0xFFFFFFFFFFFFFF00 + static_cast<uint64_t>(25. / 32. * finets);
+  // Should I read these in at m_first_time and store in static member variable or read in each time?
```
Comment (lhuth) [inline user/tlu/module/src/TluRawEvent2StdEventConverter.cc:40]: Have you checked what the penalty in computing time is? In any case, please remove the comment :)

File: user/tlu/module/src/TluRawEvent2StdEventConverter.cc [hunk 3]
```diff
@@ -29,41 +29,142 @@ bool TluRawEvent2StdEventConverter::Converting(eudaq::EventSPC d1, eudaq::Standa
...
+
+  // forward original tags:
+  d2->SetTag("FINE_TS0", d1->GetTag("FINE_TS0", "0")); // forward original tag
+  d2->SetTag("FINE_TS1", d1->GetTag("FINE_TS1", "0")); // forward original tag
+  d2->SetTag("FINE_TS2", d1->GetTag("FINE_TS2", "0")); // forward original tag
+  d2->SetTag("FINE_TS3", d1->GetTag("FINE_TS3", "0")); // forward original tag
+  d2->SetTag("FINE_TS4", d1->GetTag("FINE_TS4", "0")); // forward original tag
+  d2->SetTag("FINE_TS5", d1->GetTag("FINE_TS5", "0")); // forward original tag
+
+  // calculate (delayed) fine timestamps in ns:
```
Comment (lhuth) [inline user/tlu/module/src/TluRawEvent2StdEventConverter.cc:140]: I'm not sure if all this information is really necessary... Maybe we can make this optional? Let's see what @eyiliu 's opinion on that topic is.

File: user/tlu/module/src/TluRawEvent2StdEventConverter.cc (hunk 2 above)
Comment (simonspa) [reply user/tlu/module/src/TluRawEvent2StdEventConverter.cc:40]: @jekroege fix comment

Haven't checked penalty but it's offline, so it shouldn't really matter. The whole converter concept with static instances is a bit flawed anyway... ;)

Comment (jekroege) [reply user/tlu/module/src/TluRawEvent2StdEventConverter.cc:40]: Yep, just saw it, too...will  be fixed in the next MR I'm pushing in 5min :)

File: user/tlu/module/src/TluRawEvent2StdEventConverter.cc (hunk 3 above)
Comment (lhuth) [reply user/tlu/module/src/TluRawEvent2StdEventConverter.cc:142]: @eyiliu - comments?

**Review by lhuth — APPROVED** [review]

**Pull request merged**